src/main/java/by/dragonsurvivalteam/dragonsurvival/mixins/VaultBlockEntityServerMixin.java

```diff
@@ -0,0 +1,30 @@
+package by.dragonsurvivalteam.dragonsurvival.mixins;
+
```

Comment:
What about tryInsertKey? At the moment it will just remove the currently configured key item from the inventory rather than the new dragon key. Also, will this effect all vaults? We don't want players to be able to open normal vaults in the trial chambers with dragon keys.

Reply:
The vault will only activate if the loot table length > 0.  It also uses a tag on the key itself to check against the vault to see if that key is able to open that particular vault type.  If the tag is not present, it proceeds to the normal logic.

The vault ignores the key type when deducting keys from your inventory.  It only checks the count on the configured key's ItemStack to see how many to remove.